volume/ceph/ceph.go

```diff
@@ -34,20 +37,37 @@ func (r *Root) Create(name string) (volume.Volume, error) {
 	v, exists := r.volumes[name]
 	if !exists {
 		//TODO: Might want to map with --options rw/ro here, but then we need to sneak in the RW flag somehow
-		cmd := exec.Command("rbd", "map", name)
 		var stdout bytes.Buffer
 		var stderr bytes.Buffer
+
+		cmd := exec.Command("rbd", "create", name, "--size", strconv.Itoa(CephImageSizeMB))
 		cmd.Stdout = &stdout
 		cmd.Stderr = &stderr
-		var mappedDevicePath string
 		if err := cmd.Run(); err == nil {
-			mappedDevicePath = strings.TrimRight(stdout.String(), "\n")
-			logrus.Infof("Succeeded in mapping Ceph volume %s to %s", name, mappedDevicePath)
-		} else {
-			msg := fmt.Sprintf("Failed to map Ceph volume %s: %s - %s", name, err, strings.TrimRight(stderr.String(), "\n"))
+			logrus.Infof("Created Ceph volume %s", name)
+			mappedDevicePath, err := mapCephVolume(name)
+			if err != nil {
+				return nil, err
+			}
+			cmd = exec.Command("mkfs.ext4", "-m0", mappedDevicePath)
+			logrus.Infof("Creating ext4 fs in volume %s", mappedDevicePath)
+			if err := cmd.Run(); err != nil {
+				return nil, err
+			}
+			if err := unmapCephVolume(name, mappedDevicePath); err != nil {
```

Comment:
Why unmap and then map again? I suggest removing this unmap, and putting the below map into an else.